sdk/core/azure-core/inc/response.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <http/http.hpp>
+
+namespace Azure { namespace Core {
+
+  template <class T> class Response {
+    T m_value;
+    std::unique_ptr<Http::RawResponse> m_rawResponse;
+
+  public:
+    // Require a raw response to create a Response T
+    explicit Response(T const initialValue, std::unique_ptr<Http::RawResponse>&& rawResponse)
```

Comment:
The main goal here is to avoid copying initialValue, so would it be enough by having:
```cpp
explicit Response(T initialValue) :m_value(std::move(initialValue){}
```
??

Reply:
Clear and unambiguous suggestion:
```suggestion
    explicit Response(T initialValue, std::unique_ptr<Http::RawResponse>&& rawResponse)
```